tests/test_checkpoint.py

```diff
@@ -79,13 +67,53 @@ def test_gcs_estimator_checkpoint_save():
         MagicMock(return_value=mocked_gcs_client),
     ):
         with patch(
-            "datasetinsights.storage.checkpoint.EstimatorCheckpoint",
+            "datasetinsights.storage.checkpoint.LocalEstimatorWriter",
             MagicMock(return_value=mocked_ckpt),
         ):
-            gcs_ckpt = GCSEstimatorCheckpoint(cloud_path, prefix, suffix=suffix)
+            gcs_ckpt = GCSEstimatorWriter(cloud_path, prefix, suffix=suffix)
             gcs_ckpt.save(estimator)
 
             mocked_ckpt.save.assert_called_once_with(estimator, None)
             mocked_gcs_client.upload.assert_called_once_with(
                 path, bucket, object_key
             )
+
+
+def test_create_writer():
+    mock_local_writer = Mock()
+    with patch(
+        "datasetinsights.storage.checkpoint.LocalEstimatorWriter",
+        MagicMock(return_value=mock_local_writer),
+    ):
+        writer = EstimatorCheckpoint._create_writer("/path/to/folder", "abc")
+
+        assert writer == mock_local_writer
+
+    mock_gcs_writer = Mock()
+    with patch(
+        "datasetinsights.storage.checkpoint.GCSEstimatorWriter",
+        MagicMock(return_value=mock_gcs_writer),
+    ):
+        writer = EstimatorCheckpoint._create_writer("gs://abucket/path", "def")
+
+        assert writer == mock_gcs_writer
+
+
+@patch("datasetinsights.storage.checkpoint.load_from_http")
```

Comment:
Why is mocking required as `_get_loader_from_path` only returns method? `load_local`, `load_from_gcs`, ` load_from_http` will not actually run. You should test directly if the desired method is returned by `_get_loader_from_path`